crates/pop-cli/src/commands/build/mod.rs

```diff
@@ -62,19 +66,26 @@ impl Command {
 		#[cfg(feature = "contract")]
 		if pop_contracts::is_supported(args.path.as_deref())? {
 			// All commands originating from root command are valid
-			BuildContractCommand { path: args.path, release: args.release, valid: true }
-				.execute()?;
+			let release = match args.profile {
+				Some(profile) => profile.into(),
+				None => args.release,
+			};
+			BuildContractCommand { path: args.path, release, valid: true }.execute()?;
 			return Ok("contract");
 		}
 
 		// If only parachain feature enabled, build as parachain
 		#[cfg(feature = "parachain")]
 		if pop_parachains::is_supported(args.path.as_deref())? {
+			let profile = match args.profile {
+				Some(profile) => profile,
+				None => args.release.into(),
+			};
 			// All commands originating from root command are valid
 			BuildParachainCommand {
 				path: args.path,
 				package: args.package,
-				release: args.release,
+				profile: Some(profile),
```

Comment:
The `Option` requirement should be removed when removing the deprecated parachain build and contract command.